homeassistant/components/unifi/sensor.py

```diff
@@ -185,3 +236,172 @@ async def options_updated(self) -> None:
+        ),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=LOCAL_TEMPERATURE_SENSOR,
+        name="Local Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="temperatures",
+        value_fn=lambda device: next(
+            (
+                temperature["value"]
+                for temperature in device.raw.get("temperatures", [])
+                if temperature["name"] == "Local"
+            ),
+            None,
+        ),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=PHY_TEMPERATURE_SENSOR,
+        name="PHY Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="temperatures",
+        value_fn=lambda device: next(
+            (
+                temperature["value"]
+                for temperature in device.raw.get("temperatures", [])
+                if temperature["name"] == "PHY"
+            ),
+            None,
+        ),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=FAN_LEVEL_SENSOR,
+        name="Fan Level",
+        native_unit_of_measurement=PERCENTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="has_fan",
+        value_fn=lambda device: device.fan_level,
+    ),
+)
+
+
+class UniFiDeviceSensor(UniFiBase, SensorEntity):
+    """Base class for UniFi device sensors."""
+
+    DOMAIN = DOMAIN
+
+    entity_description: UniFiDeviceSensorEntityDescription
+
+    def __init__(
+        self,
+        device: UniFiDevice,
+        controller: UniFiController,
+        description: UniFiDeviceSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        # pylint: disable=invalid-name
+        self.TYPE = description.key
+        super().__init__(device, controller)
+        self.device = self._item
+        self.entity_description = description
+
+        self._attr_unique_id = f"{self.device.mac}_{description.key}"
+        self._attr_name = f"{self.device.name} {(description.name or '')}"
+
+    @property
+    def available(self) -> bool:
+        """Return whether the device is available."""
+        return self.controller.available and self.device.state != 0
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return a device description for the device registry."""
+        return DeviceInfo(
+            connections={(CONNECTION_NETWORK_MAC, self.device.mac)},
+            manufacturer=ATTR_MANUFACTURER,
+            model=self.device.model,
+            name=self.device.name,
+            sw_version=self.device.version,
+        )
+
+    @property
+    def native_value(self) -> Any:
```

Comment:
Make type more explicit

Reply:
Right now I could make it `float | int | datetime`, but is that any better than `Any`?

Reply:
https://github.com/home-assistant/core/blob/647a023776bedb5e1605390c91b8a89e5a68dfff/homeassistant/components/deconz/sensor.py#L345

Reply:
Oh right, I didn't know about `StateType`

Reply:
Its a big project and typing is relatively new addition to the code base